ag/validation.go

```diff
@@ -237,3 +237,14 @@ func (c *GradingCriterion) IsValid() bool {
 func (r *CourseUserRequest) IsValid() bool {
 	return r.CourseCode != "" && r.UserLogin != "" && r.CourseYear > 2019
 }
+
+func (m *Enrollments) IsValid() bool {
+	isValid := true
+	for _, e := range m.Enrollments {
+		if !e.IsValid() {
+			isValid = false
+			break
```

Comment:
No need to set `isValid = false` and `break` here. You can simply `return false`.

Actually, you don't even need the `isValid` variable because if the for-loop exits without having returned, `isValid` will be true, and thus you only need to `return m.HasCourseID()`.